compiler/sigmatch.nim

```diff
@@ -409,7 +409,7 @@ proc handleRange(f, a: PType, min, max: TTypeKind): TTypeRelation =
     elif a.kind == tyRange and
       # Make sure the conversion happens between types w/ same signedness
       (f.kind in {tyInt..tyInt64} and a[0].kind in {tyInt..tyInt64} or
-       f.kind in {tyUInt8..tyUInt32} and a[0].kind in {tyUInt8..tyInt32}) and
+       f.kind in {tyUInt8..tyUInt32} and a[0].kind in {tyUInt8..tyUInt32}) and
```

Comment:
What is the justification of this change?

Reply:
It was just a typo which broke the first version.
I can put it in a separate PR if you want

Reply:
Definitely, it's not clear what this change does. This stuff is written against the spec. Could still be wrong of course, but it's a separate PR.

Reply:
Ok, moved to #17724 (with a new protection to avoid this error in the future)